projects/packages/jetpack-mu-wpcom/src/features/100-year-plan/locked-mode.php

```diff
@@ -0,0 +1,196 @@
+ */
+function wpcom_lm_settings_page_init() {
+	if ( ! wpcom_site_has_feature( WPCOM_Features::LOCKED_MODE ) ) {
+		return;
+	}
+
+	add_settings_section(
+		'wpcom_enhanced_ownership_section',
+		__( 'Enhanced Ownership', 'jetpack-mu-wpcom' ),
+		'', // No callback needed.
+		'general'
+	);
+
+	add_settings_field(
+		'wpcom_locked_mode',
+		__( 'Locked Mode', 'jetpack-mu-wpcom' ),
+		'wpcom_locked_mode_render',
+		'general',
+		'wpcom_enhanced_ownership_section',
+		array(
+			'label_for' => 'wpcom_locked_mode',
+		)
+	);
+}
+add_action( 'admin_init', 'wpcom_lm_settings_page_init' );
+
+/**
+ * Renders the Locked Mode settings markup.
+ */
+function wpcom_locked_mode_render() {
+	?>
+	<input type="checkbox" id="wpcom_locked_mode" name="wpcom_locked_mode" <?php checked( get_option( 'wpcom_locked_mode' ) ); ?>>
+	<label for="wpcom_locked_mode"><?php esc_html_e( 'Enable Locked Mode', 'jetpack-mu-wpcom' ); ?></label>
+	<p class="description"><?php esc_html_e( 'Prevents new post and page from being created as well as existing posts and pages from being edited, and closes comments site wide.', 'jetpack-mu-wpcom' ); ?></p>
+	<?php
+}
+
+/**
+ * Removes post and page creation/editing capabilities for locked mode sites.
+ *
+ * @param array  $caps The user's capabilities.
+ * @param string $cap  The capability name.
+ * @return array
+ */
+function wpcom_lm_remove_post_capabilities( $caps, $cap ) {
+	// Return if option is not set.
+	if ( ! get_option( 'wpcom_locked_mode' ) ) {
+		return $caps;
+	}
+
+	switch ( $cap ) {
+		case 'edit_posts':
+			if ( ! defined( 'REST_API_REQUEST' ) || ! REST_API_REQUEST ) {
+				$caps = array( 'do_not_allow' );
+				break;
+			}
+
+			// This is only called on REST API requests on sites that have the Locked Mode feature, where Locked Mode is enabled.
+			$trace = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS, 8 ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions
+
+			// phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedIf
+			if ( isset( $trace[7]['class'] ) && 'WPCOM_JSON_API_Site_Settings_Endpoint' === $trace[7]['class'] && 'get_settings_response' === $trace[7]['function'] ) {
+				/*
+				 * This targets the edit_posts check in WPCOM_JSON_API_Site_Settings_Endpoint::get_settings_response().
+				 * We still want to allow the endpoint to return the settings for the site, even if the site is in locked mode.
+				 * So in this case we don't change the capabilities, for all other cases we do.
+				 */
+			} else {
+				$caps = array( 'do_not_allow' );
+			}
+			break;
+
+		case 'install_themes':
+		case 'switch_themes':
+		case 'edit_themes':
+		case 'delete_themes':
+		case 'edit_theme_options':
+		case 'customize':
+		case 'install_plugins':
+		case 'activate_plugins':
+		case 'edit_plugins':
+		case 'delete_plugins':
+		case 'edit_files':
+		case 'upload_files':
+		case 'edit_comment':
+		case 'moderate_comments':
+		case 'manage_categories':
+		case 'manage_links':
+		case 'import':
+		case 'edit_posts':
```

Comment:
'edit_posts' appears twice.

Reply:
Thanks, fixed in https://github.com/Automattic/jetpack/pull/33081/commits/165fcf11985b90dffe421d375eb4393952ea58ec